src/witan/population/lookups/lad_to_ctyua.clj

```diff
@@ -0,0 +1,146 @@
+(ns witan.population.lookups.lad-to-ctyua
+  "Lookups mapping UK Local Authority Districts to County and Unitary Authority.
+   Data from: https://geoportal.statistics.gov.uk/search?q=LUP_LTLA_UTLA.
+   Note dataset columns (<=2023) are (re)named using the aliases: LTLA->LAD & UTLA->CTYUA."
+  (:require [clojure.java.io :as io]
+            [clojure.string :as str]
+            [tech.v3.dataset :as ds]
+            [tablecloth.api :as tc]))
+
+(def LAD21_CTYUA21_EW_LU-resource-file-name
+  "Name of resource file mapping 2021 Local Authority Districts in England & Wales to County and Unitary Authority.
+   From: https://geoportal.statistics.gov.uk/datasets/ons::local-authority-district-to-county-and-unitary-authority-april-2021-lookup-in-ew/about"
+  "LUP_LAD_CTYUA/Lower_Tier_Local_Authority_to_Upper_Tier_Local_Authority_(April_2021)_Lookup_in_England_and_Wales.csv")
+
+(def LAD22-CTYUA22-EW-LU-resource-file-name
+  "Name of resource file mapping 2022 Local Authority Districts in England & Wales to County and Unitary Authority.
+   From: https://open-geography-portalx-ons.hub.arcgis.com/datasets/ons::local-authority-district-to-county-and-unitary-authority-december-2022-lookup-in-ew/about"
+  "LUP_LAD_CTYUA/Lower_Tier_Local_Authority_to_Upper_Tier_Local_Authority_(December_2022)_Lookup_in_England_and_Wales.csv")
+
+(def LAD23-CTYUA23-EQ-LU-resource-file-name
+  "Name of resource file mapping 2023 Local Authority Districts in England & Wales to County and Unitary Authority.
+   From: https://geoportal.statistics.gov.uk/datasets/ons::local-authority-district-to-county-and-unitary-authority-april-2023-lookup-in-ew/about"
+  "LUP_LAD_CTYUA/Local_Authority_District_to_County_and_Unitary_Authority_(April_2023)_Lookup_in_EW.csv")
+
+(def LAD24-CTYUA24-EQ-LU-resource-file-name
+  "Name of resource file mapping 2024 Local Authority Districts in England & Wales to County and Unitary Authority.
+   From: https://geoportal.statistics.gov.uk/datasets/ons::local-authority-district-to-county-and-unitary-authority-december-2024-lookup-in-ew/about"
+  "LUP_LAD_CTYUA/Local_Authority_to_County_and_Unitary_Authority_(December_2024)_Lookup_in_EW.csv")
+
+(def LAD25-CTYUA25-EQ-LU-resource-file-name
+  "Name of resource file mapping 2025 Local Authority Districts in England & Wales to County and Unitary Authority.
+   From: https://geoportal.statistics.gov.uk/datasets/ons::local-authority-district-to-county-and-unitary-authority-april-2025-lookup-in-ew-v2/about"
+  "LUP_LAD_CTYUA/Local_Authority_District_to_County_and_Unitary_Authority_(April_2025)_Lookup_in_EW_v2.csv")
+
+(def resource-file-name-for-year
+  "Map mapping Geogrpahy year YY strings to the corresponding resource file name."
+  {"21" LAD21_CTYUA21_EW_LU-resource-file-name
+   "22" LAD22-CTYUA22-EW-LU-resource-file-name
+   "23" LAD23-CTYUA23-EQ-LU-resource-file-name
+   "24" LAD24-CTYUA24-EQ-LU-resource-file-name
+   "25" LAD25-CTYUA25-EQ-LU-resource-file-name})
```

Comment:
:cool: I think I might need to go back to 2018, but I can add those in if I need them.